### PR TITLE
Added support for Throttling and Rate Limiting

### PR DESCRIPTION
**Pull request by ddragosd**

### CODE REVIEW

**Comment by cristianconstantin** [comment]
👍

File: init.sh [hunk 1]
```diff
@@ -30,6 +30,28 @@ sleep_duration=${MARATHON_POLL_INTERVAL:-5}
...
+    zmq_adaptor_cmd="${zmq_adaptor_cmd} -d"
+fi
+
+$zmq_adaptor_cmd >> /dev/stderr &
+sleep 3s
+# allow interprocess communication by allowing api-gateway processes to write to the socket
+chmod 777 /tmp/nginx_queue_listen
+chmod 777 /tmp/nginx_queue_push
+# keep adaptor running using a simple loop
+while true; do zmq_pid=$(ps aux | grep api-gateway-zmq-adaptor | grep -v grep) || ( echo "Restarting api-gateway-zmq-adaptor" && $zmq_adaptor_cmd >> /dev/stderr & ); sleep 60; done &
```
Comment (ddragosd) [inline init.sh]: We need to test if the permissions for `/tmp/nginx_queue_*` remain open in case `api-gateway-zmq-adaptor` is restarted.

Comment (ddragosd) [inline init.sh]: fixed the permissions issue by `chown nginx-api-gateway:nginx-api-gateway /tmp/nginx_queue_*`

**Comment by selfxp** [comment]
👍

**Pull request merged**